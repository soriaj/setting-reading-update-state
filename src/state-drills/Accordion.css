* {
   box-sizing: border-box;
}
.react_accordion{
   display: flex;
   flex-direction: column;
   line-height:50px;
}
.react_accordion > li {
   list-style: none;
}
.list_item{
   font-weight: 600;
}
button {
   background-color: #269CE9;
   border-color: #269CE9;
   border-radius: 5px;
   padding: 4px 12px;
   color: white;
}
button:hover{
   background-color: #70B9E8;
   transform: scale(1.05);
}