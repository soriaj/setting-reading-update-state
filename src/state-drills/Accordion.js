import React from 'react';
import './Accordion.css'

class Accordion extends React.Component{
   constructor(props){
      super(props)
   }
   static defaultProps = {
      sections: []
   };
   state = {
      activeSection: false,
   }
   handleButtonClick(index){
      this.setState({
         activeSection: index
      })
   }
   renderList(){
      const { sections } = this.props
      const { activeSection } = this.state
      let data = sections.map((title, index) => (
         <li className='list_item' key={index}>
            <button type='button' onClick={() => this.handleButtonClick(index)}>
               {title.title}
            </button>
            {(activeSection === index) && <p>{title.content}</p>}
         </li>
      ))
      return data
   }
   render() {
      return (
         <ul className='react_accordion'>
            {this.renderList()}
         </ul>
      )
   }
}

export default Accordion;