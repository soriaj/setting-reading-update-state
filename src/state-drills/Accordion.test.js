import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Accordion Component', () => {
   const sections = [
      {
        title: 'Section 1',
        content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
      },
      {
        title: 'Section 2',
        content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
      },
      {
        title: 'Section 3',
        content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
      },
    ];

   it('Snapshot Test 01: component renders an empy li when sections prop not supplied.', () => {
      const wrapper = shallow(<Accordion />)
      expect(toJson(wrapper)).toMatchSnapshot()
   })
   
   it('Snapshot Test 02: renders no section as active by default.', () => {
      const div = document.createElement('div');
      const wrapper = shallow(<Accordion />, div)
      expect(toJson(wrapper)).toMatchSnapshot()
      
   })

   it('Snapshot Test 03: component opens a clicked section.', () => {
      const wrapper = shallow(<Accordion sections={sections} />)
      wrapper.find('button').at(2).simulate('click')
      expect(toJson(wrapper)).toMatchSnapshot()
   })
   
   it('Snapshot Test 04: component opens last section when multiple sections have been clicked.', () => {
      const wrapper = shallow(<Accordion sections={sections} />)
      wrapper.find('button').at(0).simulate('click')
      wrapper.find('button').at(1).simulate('click')
      expect(toJson(wrapper)).toMatchSnapshot()
   })
})