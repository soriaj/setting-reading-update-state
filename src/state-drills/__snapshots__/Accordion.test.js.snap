// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component Snapshot Test 01: component renders an empy li when sections prop not supplied. 1`] = `
<ul
  className="react_accordion"
/>
`;

exports[`Accordion Component Snapshot Test 02: renders no section as active by default. 1`] = `
<ul
  className="react_accordion"
/>
`;

exports[`Accordion Component Snapshot Test 03: component opens a clicked section. 1`] = `
<ul
  className="react_accordion"
>
  <li
    className="list_item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="list_item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="list_item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion Component Snapshot Test 04: component opens last section when multiple sections have been clicked. 1`] = `
<ul
  className="react_accordion"
>
  <li
    className="list_item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="list_item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="list_item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;
